### PR TITLE
refactor: node command

### PR DESCRIPTION
**Pull request by 0xMelkor**
I made a refactor to have a more readable code. 
I've not touched the behavior, but just extracted functions and added some logging.

Hope this helps to mitigate complexity.

### CODE REVIEW

**Comment by 0xMelkor** [comment]
Hi @onbjerg, I realized too late that you opened #1079.
Feel free to drop this if necessary.

Cheers

File: bin/reth/src/node/mod.rs [hunk 1]
```diff
@@ -84,34 +86,77 @@ impl Command {
...
     pub async fn execute(self) -> eyre::Result<()> {
+        info!(target: "reth::cli", "reth {} starting", crate_version!());
+
         // Raise the fd limit of the process.
         // Does not do anything on windows.
         raise_fd_limit();
 
-        let mut config: Config =
-            confy::load_path(&self.config).wrap_err("Could not load config")?;
+        info!(target: "reth::cli", path = %self.db, "Loading configuration");
```
Comment (onbjerg) [inline bin/reth/src/node/mod.rs:95]:
```suggestion

```

File: bin/reth/src/node/mod.rs [hunk 2]
```diff
@@ -84,34 +86,77 @@ impl Command {
...
+        let mut config: Config = self.load_config()?;
+        info!(target: "reth::cli", path = %self.db, "Configuration loaded");
+
+        self.init_trusted_nodes(&mut config);
+        info!(target: "reth::cli", path = %self.db, "Trusted nodes configured");
```
Comment (onbjerg) [inline bin/reth/src/node/mod.rs:100]:
```suggestion

```

File: bin/reth/src/node/mod.rs [hunk 3]
```diff
@@ -84,34 +86,77 @@ impl Command {
...
+
+        info!(target: "reth::cli", path = %self.db, "Opening database");
+        let db = Arc::new(init_db(&self.db)?);
+        info!(target: "reth::cli", "Database opened");
+
+        self.start_metrics_endpoint()?;
+
+        info!(target: "reth::cli", "Checking genesis block");
+        let genesis = init_genesis(db.clone(), self.chain.clone())?;
+        info!(target: "reth::cli", ?genesis, "Inserted genesis");
```
Comment (onbjerg) [inline bin/reth/src/node/mod.rs:110]:
```suggestion
        let genesis = init_genesis(db.clone(), self.chain.clone())?;
```

File: bin/reth/src/node/mod.rs [hunk 4]
```diff
@@ -84,34 +86,77 @@ impl Command {
...
+
+        info!(target: "reth::cli", "Initializing consensus");
```
Comment (onbjerg) [inline bin/reth/src/node/mod.rs:112]:
```suggestion

```

File: bin/reth/src/node/mod.rs [hunk 5]
```diff
@@ -84,34 +86,77 @@ impl Command {
...
+        let consensus = self.init_consensus()?;
+        info!(target: "reth::cli", "Consensus initialized");
```
Comment (onbjerg) [inline bin/reth/src/node/mod.rs:114]:
```suggestion
        info!(target: "reth::cli", "Consensus engine initialized");
```

File: bin/reth/src/node/mod.rs [hunk 6]
```diff
@@ -84,34 +86,77 @@ impl Command {
...
+
+    fn load_config(&self) -> eyre::Result<Config> {
+        confy::load_path::<Config>(&self.config).wrap_err("Could not load config")
+    }
+
+    fn init_trusted_nodes(&self, config: &mut Config) {
         config.peers.connect_trusted_nodes_only = self.network.trusted_only;
 
         if !self.network.trusted_peers.is_empty() {
             self.network.trusted_peers.iter().for_each(|peer| {
```
Comment (onbjerg) [inline bin/reth/src/node/mod.rs:144]:
```suggestion
        if !self.network.trusted_peers.is_empty() {
            info!(target: "reth::cli", "Adding trusted nodes");
            self.network.trusted_peers.iter().for_each(|peer| {
```

**Review by onbjerg — APPROVED** [review]
Lowering the verbosity of the logs, most of these are instant, and some of the logs should just be moved to their respective functions. Ok to merge in spite of the node builder PR since it will take a while

**Pull request merged**